src/pages/home/report/ReportActionItem.js

```diff
@@ -95,105 +96,89 @@ const propTypes = {
-            this.props.translate !== nextProps.translate
-        );
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const textInputRef = useRef();
+    const popoverAnchorRef = useRef();
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.draftMessage || !this.props.draftMessage) {
+    const isDraftEmpty = !props.draftMessage;
+    useEffect(() => {
+        if (isDraftEmpty) {
             return;
         }
 
         // Only focus the input when user edits a message, skip it for existing drafts being edited of the report.
         // There is an animation when the comment is hidden and the edit form is shown, and there can be bugs on different mobile platforms
         // if the input is given focus in the middle of that animation which can prevent the keyboard from opening.
-        focusTextInputAfterAnimation(this.textInput, 100);
-    }
+        focusTextInputAfterAnimation(textInputRef.current, 100);
+    }, [isDraftEmpty]);
 
-    checkIfContextMenuActive() {
-        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(this.props.action.reportActionID)});
-    }
+    const toggleContextMenuFromActiveReportAction = useCallback(() => {
+        setIsContextMenuActive(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    }, [props.action.reportActionID]);
 
     /**
      * Show the ReportActionContextMenu modal popover.
      *
      * @param {Object} [event] - A press event.
      */
-    showPopover(event) {
-        // Block menu on the message being Edited or if the report action item has errors
-        if (this.props.draftMessage || !_.isEmpty(this.props.action.errors)) {
-            return;
-        }
-
-        this.setState({isContextMenuActive: true});
-
-        const selection = SelectionScraper.getCurrentSelection();
-        ReportActionContextMenu.showContextMenu(
-            ContextMenuActions.CONTEXT_MENU_TYPES.REPORT_ACTION,
-            event,
-            selection,
-            this.popoverAnchor,
-            this.props.report.reportID,
-            this.props.action,
-            this.props.draftMessage,
-            undefined,
-            this.checkIfContextMenuActive,
-            ReportUtils.isArchivedRoom(this.props.report),
-            ReportUtils.chatIncludesChronos(this.props.report),
-            this.props.action.childReportID,
-        );
-    }
+    const showPopover = useCallback(
+        (event) => {
+            // Block menu on the message being Edited or if the report action item has errors
+            if (props.draftMessage || !_.isEmpty(props.action.errors)) {
+                return;
+            }
+
+            setIsContextMenuActive(true);
+
+            const selection = SelectionScraper.getCurrentSelection();
+            ReportActionContextMenu.showContextMenu(
```

Comment:
Not handling focus removal from already focussed textinputs while showing context menu resulted in #44042